### DISCUSSION TITLE
Monorepo TS extensions

### DISCUSSION THREAD

**Discussion by keesvanbemmel** [directus/directus#11105] [Feature Requests]
Hi! (especially @nickrum since @rijkvanzanten told me you were the guy to ask ;) )

I'm looking for a setup/example of a local development environment where we can develop extensions in TS (using scaffolding from the extensions-sdk) but also having shared/centralized services all extensions can use.

Any pointes/repos or tips?

Thanks!

**Comment by rijkvanzanten** [comment]
Cc @erondpowell for finding a good spot in the docs 🙂

**Comment by nickrum** [comment]
Hoi @keesvanbemmel!

Unfortunately, there aren't any examples yet. I would recommend setting up a monorepo using something like npm workspaces. You could then scaffold your extensions as separate packages inside the monorepo.

For shared code, I would set up a single `shared` package for any code that is used by multiple extensions. This is very similar to the way how code is structured inside the main Directus repo. The only downside to this approach is that there is currently no way to actually share code when your extensions are deployed, so the shared code has to bundled into all of the extensions that use it.

**Comment by keesvanbemmel** [comment]
I was already thinking about setting up a [nx](https://nx.dev) environment for this.

Not being able to share code and bundle it with every extension is not something I'd want though, as these shared services tend to be large. Let's wait and see if tinkering around with NX gives me some sort of solution for that.

Will keep you posted. Timeline unknown ;)

**Comment by liamdefty** [comment]
@keesvanbemmel Did you every get to experimenting with this? :)

 I've been thinking of creating an nx plugin for directus and I'm looking for anyone who has any experience to give their input

**Reply by keesvanbemmel** [reply]
Hi @liamdefty! I experimented with this and a proof of principal was running ok. However, my knowledge of NX was not sufficient enough to set this up correctly.

I'd be more than happy to help with this if needed.

Cheers!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
